doc/themes/the-things-stack/assets/js/theme.js

```diff
@@ -12,7 +12,33 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-document.addEventListener('DOMContentLoaded', function() {
+// When you change the domain, the easiest way to redirect people without downtime is to embed redirection script inside
+// the webpage, since it is delivered 'as a whole'
+
+(function(canonical, alternatives) {
+  if (!alternatives.some((alternative) => alternative.test(document.location.href))) {
+    let url = new URL(document.location.href)
+    window.location.replace(canonical + url.pathname + url.search + url.hash)
+  }
+})('https://www.thethingsindustries.com/docs', [
+  /^https:\/\/www.thethingsindustries.com\/docs/,
+  /^https:\/\/preview.thethingsindustries.com\/docs/,
+  /^http:\/\/localhost:1313/, // local development
+  /^https:\/\/[a-zA-Z0-9]+\.tk/, // .tk domains are useful for testing, no production service depends on this
```

Comment:
Really?

What is this part for actually?

Reply:
indeed. I understand this is only a temporary solution until we have a `301` in place in `thethingsstack.io` ?

Reply:
indeed. I understand this is only a temporary solution until we have a `301` in place in `thethingsstack.io` ?

Reply:
Without this, after merge people will see broken docs until we have `301` in place of `thethingsstack.io`. Can we coordinate these two things, in particular when DNS has multiple layers of caching?

Reply:
Let's not do this please.
Also this is a bad idea
` /^https:\/\/[a-zA-Z0-9]+\.tk/`
We should not allow random wildcard domains. We don't control the entire DNS space of `*.tk`.